### PR TITLE
Fix HTTPProbe initialize header

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?
/sig node
/kind bug

#### What this PR does / why we need it:

The order of initializing the headers is wrong.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by adisky** [comment]
unrelated just testing this job
/test pull-containerd-node-e2e

**Comment by shubheksha** [comment]
@wzshiming could you add some more details to the PR body? Is there an existing issue for this? Could you also add a test?

File: pkg/probe/http/http.go [hunk 1]
```diff
@@ -94,10 +94,10 @@ func DoHTTPProbe(url *url.URL, headers http.Header, client GetHTTPInterface) (pr
 		// Convert errors into failures to catch timeouts.
 		return probe.Failure, err.Error(), nil
 	}
+	if headers == nil {
```
Comment (SergeyKanzhelev) [inline pkg/probe/http/http.go:97]: it is not really necessary as the if statement will work without issue even when `headers` is `nil`.

Comment (SergeyKanzhelev) [reply pkg/probe/http/http.go:97]: this is on the edge of being a trivial unnecessary edit. I agree the code is better this way so it's lgtm

/lgtm
/hold cancel

**Pull request merged**